model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -65,6 +67,69 @@ def find_and_replace_pattern(self, graph: Graph):
+                producer_port = consumer_port.get_source()
+                producer_shape = producer_port.data.get_shape()
+
+                producer_data_node = eltwise_node.in_node(in_port_idx)
+                edge_attrs = graph.get_edge_data(producer_data_node.id, eltwise_node.id)[0]
+                if is_first_pass and 'unsqueeze_dims' in edge_attrs and len(edge_attrs['unsqueeze_dims']) > 0:
+                    unsqueeze_dims = tuple([x for x in edge_attrs['unsqueeze_dims']])
+                elif not is_first_pass and len(producer_shape) != len(eltwise_shape):
+                    unsqueeze_dims = tuple(np.arange(len(eltwise_shape) - len(producer_shape), dtype=np.int64))
+                else:
+                    continue
+                if not producer_port in mapping:
+                    mapping.update({producer_port: {unsqueeze_dims: [consumer_port]}})
+                elif not unsqueeze_dims in mapping[producer_port]:
+                    mapping[producer_port].update({unsqueeze_dims: [consumer_port]})
+                else:
+                    mapping[producer_port][unsqueeze_dims].append(consumer_port)
+
+        # Walk through each produced in the map and insert Reshape nodes between a producer and eltwise nodes
+        for producer_port in mapping.keys():
+            producer_node = producer_port.node
+            for unsqueeze_dims in mapping[producer_port].keys():
+                unsqueeze_name = producer_node.soft_get('name', producer_node.id) + '/EltwiseReshape'
+                unsqueeze_node = create_op_with_const_inputs(graph, Unsqueeze, {1: int64_array(list(unsqueeze_dims))},
+                                                             {'name': unsqueeze_name})
+
+                unsqueeze_node.in_port(0).connect(producer_port)
+
+                # Insert Reshape with determined output shape between the current producer and eltwise node
+                for consumer_port in mapping[producer_port][unsqueeze_dims]:
+                    consumer_port.connect(unsqueeze_node.out_port(0))
+
+                # The shape and value adjustments must be explicitly done within the transformation
+                # since the transformation is called from Fusing transformation that excludes
+                # automatic call of shape inference pass
+                producer_port_value = producer_port.data.get_value()
+                producer_port_shape = producer_port.data.get_shape()
+                new_shape = producer_port_shape
+                for unsqueeze_dim in unsqueeze_dims:
+                    new_shape = np.insert(new_shape, unsqueeze_dim, 1)
+                if producer_port_value is not None:
+                    unsqueeze_node.out_port(0).data.set_value(np.reshape(producer_port_value, new_shape))
+                else:
+                    unsqueeze_node.out_port(0).data.set_shape(new_shape)
+
+
+class EltwiseInputReshapeFirstPass(MiddleReplacementPattern):
```

Comment:
Can we rename the transformation to be more meaningful? "FirstPass" is not intuitive...